modules/build-tarball.nix

```diff
@@ -23,6 +23,19 @@ let
       wsl.defaultUser = "nixos";
       ${lib.optionalString (!config.wsl.nativeSystemd) "wsl.nativeSystemd = false;"}
 
+      system.activationScripts.NixOS-WSL-Permissions-Fixer.text =
+      '''
+        if [ -f /etc/nixos/.permissions-fixer-needed ]; then
+          echo "[NixOS-WSL-Permissions-Fixer] Making /etc/nixos writable to current user..."
+          ${pkgs.acl}/bin/setfacl -R -m u:nixos:rwx /etc/nixos
```

Comment:
Giving executable permissions to it is fine or it may be a security risk?

Reply:
Explicitly setting the executable bit shouldn't be necessary. The directory should have it set already (needed to ls it) and the files in there don't need it.
Two other things: Please move the whole thing to `wsl-distro.nix` instead of the config template. Besides that i'd rather have the files writeable by root instead of the default user (who might have a user name other than nixos anyway). With that using the `acl` package shouldn't be necessary. `${pkgs.coreutils}/chmod -R u+w` would be sufficient.
Rest looks good to me

Reply:
> Besides that i'd rather have the files writeable by root instead of the default user (who might have a user name other than nixos anyway). With that using the `acl` package shouldn't be necessary. `${pkgs.coreutils}/chmod -R u+w` would be sufficient. Rest looks good to me

Files is already writable by root (tested in lastest release), I wonder what we are trying to fix then?

Reply:
You're right, I just double-checked. I was convinced it was still `-r--r--r--`, but apparently that _was_ fixed already. I am so sorry for having wasted your time...

Reply:
Don't worry about that, it helped me learn some very cool stuff!